handlers/dashboard.go

```diff
@@ -344,49 +349,89 @@ func DashboardDataProposals(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func DashboardWithdrawals(w http.ResponseWriter, r *http.Request) {
-
-	// [WIP]
+func DashboardDataWithdrawals(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", "application/json")
 
 	q := r.URL.Query()
 	validatorLimit := getUserPremium(r).MaxValidators
-	filterArr, err := parseValidatorsFromQueryString(q.Get("validators"), validatorLimit)
+	validators, err := parseValidatorsFromQueryString(q.Get("validators"), validatorLimit)
 	if err != nil {
 		http.Error(w, "Invalid query", 400)
 		return
 	}
-	filter := pq.Array(filterArr)
 
-	proposals := []struct {
-		Slot   uint64
-		Status uint64
-	}{}
+	draw, err := strconv.ParseUint(q.Get("draw"), 10, 64)
+	if err != nil {
+		logger.Errorf("error converting datatables data parameter from string to int: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+	start, err := strconv.ParseUint(q.Get("start"), 10, 64)
+	if err != nil {
+		logger.Errorf("error converting datatables start parameter from string to int: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
 
-	err = db.ReaderDb.Select(&proposals, `
-		SELECT slot, status
-		FROM blocks
-		WHERE proposer = ANY($1)
-		ORDER BY slot`, filter)
+	orderColumn := q.Get("order[0][column]")
+	orderByMap := map[string]string{
+		"0": "validatorindex",
+		"1": "block_slot",
+		"2": "block_slot",
+		"3": "withdrawalindex",
+		"4": "address",
+		"5": "amount",
+	}
+	orderBy, exists := orderByMap[orderColumn]
+	if !exists {
+		orderBy = "validatorindex"
+	}
+	orderDir := q.Get("order[0][dir]")
+	if orderDir != "asc" {
+		orderDir = "desc"
+	}
+
+	length := uint64(10)
+
+	withdrawalCount, err := db.GetDashboardWithdrawalsCount(validators)
 	if err != nil {
-		logger.WithError(err).WithField("route", r.URL.String()).Error("error retrieving block-proposals")
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		logger.Errorf("error retrieving dashboard validator withdrawals count: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
 		return
 	}
 
-	proposalsResult := make([][]uint64, len(proposals))
-	for i, b := range proposals {
-		proposalsResult[i] = []uint64{
-			uint64(utils.SlotToTime(b.Slot).Unix()),
-			b.Status,
-		}
+	withdrawals, err := db.GetDashboardWithdrawals(validators, length, start, orderBy, orderDir)
+	if err != nil {
+		logger.Errorf("error retrieving validator withdrawals: %v", err)
```

Comment:
I would suggest to always use utils.LogError / utils.LogFatal, which will do all that for you.